src/runtimes/wasi_cli/errors.rs

```diff
@@ -1,23 +1,55 @@
 use thiserror::Error;
 
+pub type Result<T> = std::result::Result<T, WasiRuntimeError>;
+
 #[derive(Error, Debug)]
 pub enum WasiRuntimeError {
     #[error("cannot set wasi args")]
-    WasiStringArray(#[from] wasi_common::StringArrayError),
+    WasiStringArray(#[source] wasi_common::StringArrayError),
 
     #[error("program exited with code {code:?}; stderr set to '{stderr}', error: '{error}'")]
     WasiEvaluation {
         code: Option<i32>,
         stderr: String,
+        #[source]
         error: wasmtime::Error,
     },
 
-    #[error("cannot instantiate module: {0}")]
-    WasmInstantiate(wasmtime::Error),
-
     #[error("cannot find `_start` function inside of module: {0}")]
-    WasmMissingStartFn(wasmtime::Error),
+    WasmMissingStartFn(#[source] wasmtime::Error),
 
     #[error("{name} pipe conversion error: {error}")]
     PipeConversion { name: String, error: String },
+
+    #[error("cannot instantiate module: {0}")]
+    WasmLinkerError(#[source] wasmtime::Error),
+
+    #[error("cannot add to linker: {0}")]
+    WasmInstantiate(#[source] wasmtime::Error),
+
+    #[error("cannot find 'mem' export")]
+    WasiMemExport(),
+
+    #[error("'mem' export cannot be converted into a Memory instance")]
+    WasiMemExportCannotConvert(),
```

Comment:
the parentheses here indicate that the error is an empty tuple variant, we can remove this and use `WasiMemExportCannotConvert` directly.
Could you change the other error variants that slipped from the past PRs?

Reply:
Doh true, done in https://github.com/kubewarden/policy-evaluator/pull/426/commits/8b41351bfb794c4e5589fd5dcd08e9165dd4c712